test/integration/local-propagation.test.ts

```diff
@@ -127,17 +128,17 @@ describe('local propagation', () => {
         await waitForCondition(() => mqttClient1.connected)
         await waitForCondition(() => mqttClient2.connected)
 
-        await mqttClient1.subscribe(freshStreamId)
-        await mqttClient2.subscribe(freshStreamId)
-
-        mqttClient1.on('message', (topic, message) => {
+        mqttClient1.on('message', (_topic, message) => {
             client1Messages.push(JSON.parse(message.toString()))
         })
 
-        mqttClient2.on('message', (topic, message) => {
+        mqttClient2.on('message', (_topic, message) => {
             client2Messages.push(JSON.parse(message.toString()))
         })
 
+        await mqttClient1.subscribe(freshStreamId)
+        await mqttClient2.subscribe(freshStreamId)
```

Comment:
Moved these out of principle, should always attach listeners before triggering the thing that could cause events to be fired.